relayer/chain/substrate/chain.go

```diff
@@ -93,3 +109,19 @@ func (ch *Chain) Stop() {
 func (ch *Chain) Name() string {
 	return Name
 }
+
+func (ch *Chain) queryEthereumInitParams() (*ethereum.HeaderID, error) {
+	storageKey, err := types.CreateStorageKey(&ch.conn.metadata, "VerifierLightclient", "FinalizedBlock", nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var headerID ethereum.HeaderID
+	_, err = ch.conn.api.RPC.State.GetStorageLatest(storageKey, &headerID)
+	if err != nil {
+		return nil, err
+	}
+
+	headerID.Number = types.NewU64(uint64(headerID.Number) + 1)
+	return &headerID, nil
```

Comment:
Again, this block should be referenced by hash, not by height, to prevent a situation where we jump to a different fork at that height.

Reply:
It's not possible to provide the hash corresponding to `headerID.Number + 1` here. The newest header that the Substrate node knows is `headerID`. It doesn't have the hashes for headers that come after.

For the purpose of syncing from an old header, this is ok. As long as the height is at least 35 blocks away from the tip, we assume whatever header is returned by `HeaderByNumber` is final (see https://github.com/Snowfork/polkadot-ethereum/blob/main/parachain/runtime/src/lib.rs#L268). Admittedly, I'm not handling the case where we get < 35 away from the tip here. I'll put that in the sync rewrite

Reply:
Sounds good. And of course we don't know future hashes ;)